### ISSUE TITLE
[java] CloseResource: False positive for allowedResourceMethodPatterns entries when using unqualified method calls

### ISSUE THREAD

**Issue by rturner-edjuster** [pmd/pmd#6502]
**Affects PMD Version:** 
PMD 7.22.0 (7f74d775143d842c9a9964e59e64284596a83a84, 2026-02-27T10:20:20Z)

**Description:**
If you add custom entries to the `allowedResourceMethodPatterns` property of the `CloseResource` rule, those functions still produce violations if they are invoked from within the same class, or from a derived class. They correctly exclude violations from unrelated classes.

**Code Sample demonstrating the issue:**

####  src/mypack/Action.java
```java
package mypack;

import java.sql.Connection;

public abstract class Action implements AutoCloseable {
    private Connection connection;

    @Override
    public void close() throws Exception {
        connection.close();
    }

    public Connection getConnection() {
        return connection;
    }

    public abstract void run();

    public void func() {
        final var c = getConnection(); // violation detected
    }
}
```

####  src/mypack/ConsumerAction.java
```java
package mypack;

public class ConsumerAction extends Action {

    @Override
    public void run() {
        final var c = getConnection(); // violation detected
    }
}
```

####  src/mypack/ConsumerClass.java
```java
package mypack;

public class ConsumerClass {

    public void run(Action action) {
        final var c = action.getConnection(); // no violation detected
    }
}
```

####  test-ruleset.xml
```xml
<ruleset xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd"
         name="Customized ruleset"
>
    <description>Customized rule set</description>
    <rule ref="category/java/errorprone.xml/CloseResource">
        <properties>
            <property name="allowedResourceMethodPatterns"
                      value="mypack.Action#getConnection()" />
        </properties>
    </rule>
</ruleset>
```

**Steps to reproduce:**

```shell
javac -d classes src/mypack/*
pmd check --no-progress --rulesets=test-ruleset.xml --dir=src --aux-classpath classes
```

**Output:**

```
[WARN] This analysis could be faster, please consider using Incremental Analysis: https://docs.pmd-code.org/pmd-doc-7.22.0/pmd_userdocs_incremental_analysis.html
src/mypack/Action.java:20:	CloseResource:	Ensure that resources like this Connection object are closed after use
src/mypack/ConsumerAction.java:7:	CloseResource:	Ensure that resources like this Connection object are closed after use
```

**Running PMD through:** *[CLI | Maven]*

Java version: 25.0.2, vendor: Amazon.com Inc., runtime: /Library/Java/JavaVirtualMachines/amazon-corretto-25.jdk/Contents/Home
